client/asset/eth/eth.go

```diff
@@ -2098,6 +2099,71 @@ func (w *TokenWallet) EstimateRegistrationTxFee(feeRate uint64) uint64 {
 	return g.Transfer * feeRate
 }
 
+// ValidateAddress checks whether the provided address is a valid hex-encoded
+// Ethereum address.
+func (w *ETHWallet) ValidateAddress(address string) bool {
+	return common.IsHexAddress(address)
+}
+
+// ValidateAddress checks whether the provided address is a valid hex-encoded
+// Ethereum address.
+func (w *TokenWallet) ValidateAddress(address string) bool {
+	return common.IsHexAddress(address)
+}
+
+// isValidSend is a helper function for both token and ETH wallet. It returns an
+// error if subtract is true, addr is invalid or value is zero.
+func isValidSend(addr string, value uint64, subtract bool) error {
+	if value == 0 {
+		return fmt.Errorf("cannot send zero amount")
+	}
+	if subtract {
+		return fmt.Errorf("wallet does not support subtracting network fee from send amount")
+	}
+	if !common.IsHexAddress(addr) {
+		return fmt.Errorf("invalid hex address %q", addr)
+	}
+	return nil
+}
+
+// EstimateSendTxFee returns a tx fee estimate for a send tx. The provided fee
+// rate is ignored since all sends will use an internally derived fee rate.
+// Also, the value provided is ignored because this implementation does not need
+// it for the fee estimation.
+func (w *ETHWallet) EstimateSendTxFee(addr string, value, _ uint64, subtract bool) (uint64, bool, error) {
+	if err := isValidSend(addr, value, subtract); err != nil && addr != "" { // fee estimate for a send tx.
+		return 0, false, err
+	}
+
+	maxFeeRate, err := w.recommendedMaxFeeRate(w.ctx)
+	if err != nil {
+		return 0, false, fmt.Errorf("error getting max fee rate: %w", err)
+	}
+	return defaultSendGasLimit * dexeth.WeiToGwei(maxFeeRate), w.ValidateAddress(addr), nil
```

Comment:
Both btc and dcr wallets will return an error for insufficient funds. Any reason that we shouldn't do the same here?

Reply:
`eth` implementation does not depend on `sendAmount`.

Reply:
I would expect insufficient balance to error consistently across wallets.

Reply:
If that's what we want it needs to be part of the `TxFeeEstimator` interface docs that an implementation must consider balance, because it's not even implied by the method name.  The other assets only consider it as a side effect of computing the fee to send a given amount because it's based on the number of inputs.  @ukane-philemon's point is that with ETH, you know the fee to send X for a given fee rate... without even having to consider a wallet at all.

This is actually one of the aspects of this current work that has been irking me.  The scope and purpose is not well defined.